server/src/main/java/org/apache/iotdb/db/mpp/execution/operator/process/join/RowBasedTimeJoinOperator.java

```diff
@@ -198,14 +207,35 @@ public TsBlock next() {
       }
       tsBlockBuilder.declarePosition();
     } while (currentTime < currentEndTime && !timeSelector.isEmpty());
-    return tsBlockBuilder.build();
+    retainedTsBlock = tsBlockBuilder.build();
+    LOGGER.info("Current tsBlock size is : {}", retainedTsBlock.getRetainedSizeInBytes());
+    return getResultTsBlock();
+  }
+
+  private TsBlock getResultTsBlock() {
+    long maxSizePerTsBlock = TSFileDescriptor.getInstance().getConfig().getMaxTsBlockSizeInBytes();
+    long maxLength = maxSizePerTsBlock / (1 + outputColumnCount);
+    TsBlock resultTsBlock;
+    if (retainedTsBlock.getPositionCount() - retainedStartOffset < maxLength) {
```

Comment:
maxLength is size in bytes, but `retainedTsBlock.getPositionCount() - retainedStartOffset` is line number, why comparing these two?